### DISCUSSION TITLE
How to use `get_text` with `textpage` argument

### DISCUSSION THREAD

**Discussion by scorphus** [pymupdf/PyMuPDF#3819] [Looking for help]
The [manual](https://pymupdf.readthedocs.io/en/latest/page.html#Page.get_text) and [source code](https://github.com/pymupdf/PyMuPDF/blob/9c19e27810a98f6a0d6652254384277525733258/src/utils.py#L750-L751) state that `textpage` can be passed to `get_text` to _“use a previously created [TextPage](https://pymupdf.readthedocs.io/en/latest/textpage.html#textpage). This reduces execution time very significantly: by more than 50% and up to 95%, depending on the extraction option. If specified, the ‘flags’ and ‘clip’ arguments are ignored, because they are textpage-only properties. If omitted, a new, temporary textpage will be created. (New in v1.19.0)”_

Can somebody please offer an example of `textpage` being passed as argument to `get_text` for a multi-page PDF?

Also, how beneficial would it be and why would a user want to pass use `get_text` in such way?

Thanks!

**Comment by JorjMcKie** [comment]
Methods `Page.get_text` and `Page.search_for` are wrappers for `TextPage` methods (see documentation there).
As such they will create a temporary `TextPage` to work with each time. While this is convenient, it also is overhead that you can avoid - see [this](https://github.com/pymupdf/PyMuPDF-Utilities/blob/master/jupyter-notebooks/testpage-performance.ipynb) Jupyter notebook. It should answer all your questions.
